atom/browser/web_contents_permission_helper.cc

```diff
@@ -55,18 +55,22 @@ WebContentsPermissionHelper::~WebContentsPermissionHelper() {
 void WebContentsPermissionHelper::RequestPermission(
     content::PermissionType permission,
     const base::Callback<void(bool)>& callback,
-    const GURL& security_origin, bool user_gesture) {
-  auto rfh = web_contents_->GetMainFrame();
+    const GURL& frame_url, bool user_gesture,
+    content::RenderFrameHost* rfh) {
+  GURL url;
+  if (!rfh)
```

Comment:
I think we should set this in `HandleExternalProtocolInUI` so the logic is all in one place and we always pass an rfh to this method